packages/compiler/src/experimental/typekit/kits/type.ts

```diff
@@ -272,5 +281,42 @@ defineKit<TypekitExtension>({
     isUserDefined(type) {
       return getLocationContext(this.program, type).type === "project";
     },
+    inNamespace(type: Type, namespace: Namespace): boolean {
+      compilerAssert(type !== undefined, "parameter 'type' is undefined");
+      compilerAssert(namespace !== undefined, "parameter 'namespace' is undefined");
+
+      // A namespace is always in itself
+      if (type === namespace) {
+        return true;
+      }
+
+      // Handle types with known containers
+      switch (type.kind) {
+        case "ModelProperty":
+          if (type.sourceProperty) {
```

Comment:
shouldn't go to sourceProperty as this might go to a different namespace. It links to the prop that was used to consutrct it in a spread or model is

Reply:
Ah, got it. Fixed!